common/web/gesture-recognizer/src/test/auto/browser/cases/recordedCoordSequences.js

```diff
@@ -0,0 +1,73 @@
+var assert = chai.assert;
+var expect = chai.expect;
+
+describe("Layer one - DOM -> InputSequence", function() {
+  this.timeout(testconfig.timeouts.standard);
+
+  before(function() {
+    fixture.setBase('');
+  });
+
+  beforeEach(function(done) {
+    fixture.load('host-fixture.html');
+    this.controller = new Testing.HostFixtureLayoutController();
+    this.controller.connect().then(() => done());
+  });
+
+  afterEach(function() {
+    this.controller.destroy();
+    fixture.cleanup();
+    fixture.cleanup();
+  });
+
+  describe('recorded input sequences', function() {
+    // We rely on this function to have the same context as `it` - the test-definition function.
+    let replayAndCompare = function(testObj) {
+      // Everything else can be extracted to a common test function... probably.
+      let playbackEngine = new Testing.InputSequenceSimulator(this.controller);
+      let result = playbackEngine.replay(testObj);
+
+      assert.equal(result.set.length, testObj.set.length);
+
+      // Removes the timestamp element; we know that this component won't match, as the playback engine
+      // doesn't care about it.
+      let sampleCleaner = (sample) => { return {targetX: sample.targetX, targetY: sample.targetY} };
+
+      // Returns just the observed, cleaned samples for a sequence object.  The recorded coordinates
+      // should match perfectly.
+      let seqCleaner = (seq) => { return seq.sequence.samples.map(sampleCleaner) };
+      let cleanOriginalSet = testObj.set.map(seqCleaner);
+      let cleanResultSet   = result .set.map(seqCleaner);
+
+      expect(cleanResultSet).to.deep.equal(cleanOriginalSet);
+
+      // The 'terminationEvent' property should match.  Any sequence that was "canceled" should still
+      // cancel; that's a pretty critical detail!
+      let terminationEventMapper = (seq) => seq.terminationEvent;
+      expect(result.set.map(terminationEventMapper)).to.deep.equal(testObj.set.map(terminationEventMapper));
+    }
+
+    // List all relevant fixtures in src/test/resources/json.
+    let testRecordings = [
+      'canaryRecording',
+      'desktopRoamAndReturn',
+      'mobileSafeZoneCancel',
+      'mobileProximityApproach',
+      'embeddedBorderCancel',
+      'hardBorderCancel',
+      'popupLongRoamingEnd',
+      'popupShimCancel',
+      'popupSafePersistence',
+      'basicMultitouch'
```

Comment:
Each of these corresponds to a user test defined on #6878; they're listed in order of their definitions on that PR.

I opted to give them more meaningful/descriptive names as unit tests.

Note that 'popupSafePersistence' (#6878's TEST_4_3_2_SAFE_ZONE) is slightly tweaked from its original version on that PR, as its written form there was a little different from the intended test.